torchvision/datasets/stanford_cars.py

```diff
@@ -0,0 +1,124 @@
+        "c3b158d763b6e2245038c8ad08e45376",
+    )  # md5 checksum for annotations
+
+    def __init__(
+        self,
+        root: str,
+        train: bool = True,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        download: bool = False,
+    ) -> None:
+
+        try:
+            import scipy.io as sio
+        except ImportError:
+            raise RuntimeError("Scipy is not found. This dataset needs to have scipy installed: pip install scipy")
+
+        super().__init__(root, transform=transform, target_transform=target_transform)
+
+        self.train = train
+
+        if download:
+            self.download()
+
+        if not self._check_exists():
+            raise RuntimeError("Dataset not found. You can use download=True to download it")
+
+        self._samples = [
+            (
+                os.path.join(self.root, f"cars_{'train' if self.train else 'test'}", annotation["fname"]),
+                annotation["class"] - 1,  # Beware stanford cars target mapping  starts from 1
+            )
+            for annotation in sio.loadmat(
+                os.path.join(
+                    self.root,
+                    *["devkit", "cars_train_annos.mat"] if self.train else ["cars_test_annos_withlabels.mat"],
+                ),
+                squeeze_me=True,
+            )["annotations"]
+        ]
+
+        class_names = sio.loadmat(os.path.join(self.root, "devkit", "cars_meta.mat"))["class_names"][0]
+        self.classes = {class_name[0]: i for i, class_name in enumerate(class_names)}
+
+    def __len__(self) -> int:
+        return len(self._samples)
+
+    def __getitem__(self, idx: int) -> Tuple[Any, Any]:
+        """Returns pil_image and class_id for given index"""
+        image_path, target = self._samples[idx]
+        pil_image = Image.open(image_path).convert("RGB")
+
+        if self.transform is not None:
+            pil_image = self.transform(pil_image)
+        if self.target_transform is not None:
+            target = self.target_transform(target)
+        return pil_image, target
+
+    def download(self) -> None:
+        if self._check_exists():
+            return
+
+        download_and_extract_archive(url=self.urls[self.train], download_root=self.root, md5=self.md5s[self.train])
+        download_and_extract_archive(url=self.annot_urls[1], download_root=self.root, md5=self.annot_md5s[1])
+        if not self.train:
+            download_url(
+                url=self.annot_urls[0],
+                root=self.root,
+                md5=self.annot_md5s[0],
+            )
+
+    def _check_exists(self) -> bool:
+        return (
+            os.path.exists(os.path.join(self.root, f"cars_{'train' if self.train else 'test'}"))
+            and os.path.isdir(os.path.join(self.root, f"cars_{'train' if self.train else 'test'}"))
+            and os.path.exists(os.path.join(self.root, "devkit", "cars_meta.mat"))
```

Comment:
As noted in https://github.com/pytorch/vision/pull/5166#discussion_r781048168, the tests still fail since the fake data generation does not include `cars_meta.mat`. You can verify locally by running `pytest test/test_datasets.py::StanfordCarsTestCase`.

Reply:
Yes i was trying to see whether tests are passing on local machine or not according to guide, but there was an error saying

`ERROR test/test_datasets.py - AttributeError: module 'torchvision.datasets' has no attribute 'Sintel'`

Thats why I wasnt able to check tests locally

Reply:
Could you post the full stack trace? Could you please also post the output of `python -c "import torchvision; print(torchvision.__version__)"`?

Reply:
Full Trace is here-

(deeppy11) C:\Users\Abhijit_asus\PycharmProjects\vision>pytest test/test_datasets.py::StanfordTestCase
================================================= test session starts =================================================
platform win32 -- Python 3.9.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: C:\Users\Abhijit_asus\PycharmProjects\vision, configfile: pytest.ini
plugins: cov-2.12.1
collected 0 items / 1 error

======================================================= ERRORS ========================================================
_______________________________________ ERROR collecting test/test_datasets.py ________________________________________
Traceback (most recent call last):
  File "C:\Users\Abhijit_asus\PycharmProjects\vision\test\test_datasets.py", line 949, in <module>
    class KineticsTestCase(datasets_utils.VideoDatasetTestCase):
  File "C:\Users\Abhijit_asus\PycharmProjects\vision\test\test_datasets.py", line 950, in KineticsTestCase
    DATASET_CLASS = datasets.Kinetics
AttributeError: module 'torchvision.datasets' has no attribute 'Kinetics'
=============================================== short test summary info ===============================================
ERROR test/test_datasets.py - AttributeError: module 'torchvision.datasets' has no attribute 'Kinetics'
================================================== 1 error in 0.09s ===================================================
ERROR: not found: C:\Users\Abhijit_asus\PycharmProjects\vision\test\test_datasets.py::StanfordTestCase
(no name 'C:\\Users\\Abhijit_asus\\PycharmProjects\\vision\\test\\test_datasets.py::StanfordTestCase' in any of [<Module test/test_datasets.py>])


and Torchvision Version is `0.10.0`

Reply:
1. To develop `torchvision` you should either build from source or install a nightly build. In your case the latter is sufficient. See https://github.com/pytorch/vision/blob/main/CONTRIBUTING.md#install-pytorch-nightly
2. You invoke `pytest` with the wrong command

    `pytest test/test_datasets.py::StanfordTestCase` (wrong)
    `pytest test/test_datasets.py::StanfordCarsTestCase` (correct)
    
    You can also do `pytest test/test_datasets.py -k cars` in which case will collect all tests that have the phrase `"cars"` in their name.

Reply:
> 1. To develop `torchvision` you should either build from source or install a nightly build. In your case the latter is sufficient. See https://github.com/pytorch/vision/blob/main/CONTRIBUTING.md#install-pytorch-nightly
> 2. You invoke `pytest` with the wrong command
>    `pytest test/test_datasets.py::StanfordTestCase` (wrong)
>    `pytest test/test_datasets.py::StanfordCarsTestCase` (correct)
>    You can also do `pytest test/test_datasets.py -k cars` in which case will collect all tests that have the phrase `"cars"` in their name.

Ohh I'm so sorry; I should have noticed this

Reply:
Regarding the version - for development; I have built from source and the version for the same is `0.12.0a0` (vision/version.txt ); So I don't think there is an issue with the built. (correct me if I am wrong)

Reply:
> 1. To develop `torchvision` you should either build from source or install a nightly build. In your case the latter is sufficient. See https://github.com/pytorch/vision/blob/main/CONTRIBUTING.md#install-pytorch-nightly
> 2. You invoke `pytest` with the wrong command
>    `pytest test/test_datasets.py::StanfordTestCase` (wrong)
>    `pytest test/test_datasets.py::StanfordCarsTestCase` (correct)
>    You can also do `pytest test/test_datasets.py -k cars` in which case will collect all tests that have the phrase `"cars"` in their name.

still it is not working; also tried with  `pytest test/test_datasets.py -k cars` 


(deeppy11) C:\Users\Abhijit_asus\PycharmProjects\vision>pytest test/test_datasets.py::StanfordCarsTestCase
================================================= test session starts =================================================
platform win32 -- Python 3.9.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: C:\Users\Abhijit_asus\PycharmProjects\vision, configfile: pytest.ini
plugins: cov-2.12.1
collected 0 items / 1 error

======================================================= ERRORS ========================================================
_______________________________________ ERROR collecting test/test_datasets.py ________________________________________
Traceback (most recent call last):
  File "C:\Users\Abhijit_asus\PycharmProjects\vision\test\test_datasets.py", line 949, in <module>
    class KineticsTestCase(datasets_utils.VideoDatasetTestCase):
  File "C:\Users\Abhijit_asus\PycharmProjects\vision\test\test_datasets.py", line 950, in KineticsTestCase
    DATASET_CLASS = datasets.Kinetics
AttributeError: module 'torchvision.datasets' has no attribute 'Kinetics'
=============================================== short test summary info ===============================================
ERROR test/test_datasets.py - AttributeError: module 'torchvision.datasets' has no attribute 'Kinetics'
================================================== 1 error in 0.02s ===================================================
ERROR: not found: C:\Users\Abhijit_asus\PycharmProjects\vision\test\test_datasets.py::StanfordCarsTestCase
(no name 'C:\\Users\\Abhijit_asus\\PycharmProjects\\vision\\test\\test_datasets.py::StanfordCarsTestCase' in any of [<Module test/test_datasets.py>])

Reply:
This happens because you are using `torchvision==0.10.0` and the `Kinetics` dataset was not available then. Please either install from source or a recent nightly build as described in the contribution guide.